webpack/assets/javascripts/react_app/common/i18nProviderWrapperFactory.js

```diff
@@ -9,31 +10,18 @@ const i18nProviderWrapperFactory = (
 ) => WrappedComponent => {
   const wrappedName = getDisplayName(WrappedComponent);
 
-  class I18nProviderWrapper extends React.Component {
-    constructor(props) {
-      super(props);
-      this.state = { i18nLoaded: false };
-
-      intl.ready.then(() => {
-        this.setState({ i18nLoaded: true });
-      });
-    }
-
-    render() {
-      if (!this.state.i18nLoaded) {
-        return <span />;
-      }
-      return (
-        <IntlProvider
-          locale={intl.locale}
-          initialNow={initialNow}
-          timeZone={timezone || intl.timezone}
-        >
-          <WrappedComponent {...this.props} />
-        </IntlProvider>
-      );
-    }
-  }
+  const I18nProviderWrapper = props => {
+    const i18nReady = useSelector(state => state.app.i18nReady);
+    return i18nReady ? (
```

Comment:
maybe for readability:
```
i18Ready && <IntlProvider....>
```

Reply:
This would return false, what I heard critic on.
Maybe this would be funcionaly the same, but more readable?
```suggestion
    return null if !i18nReady;
    <IntlProvider ...
```

Reply:
@ohadlevy @ezr-ondrej  According to the [React docs](https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator), `&&` is one of the recommended ways to do conditional rendering.  And I think it's more readable than ternary if it's more than one line.  So I think
```
return (
{i18nReady &&
<IntlProvider ... />
}
```
would be great.